File: Frontend/src/app/shared/models/basket.ts
import { Brand } from './brand';
import { Type } from './type';
import * as cuid2 from '@paralleldrive/cuid2';

export interface Basket {
  id: string;
  items: BasketItem[];
  clientSecret?: string;
  paymentIntentId?: string;
  deliveryMethodId?: number;
}

export interface BasketItem {
  id: number;
  productName: string;
  price: number;
  quantity: number;
  pictureUrl: string;
  brand: Brand;
  type: Type;
}

export class Basket implements Basket {
  id = cuid2.createId();
  items: BasketItem[] = [];
}

export interface BasketTotal {
  shipping: number;
  subtotal: number;
  total: number;
}

File: Backend/ShopNet.DAL/Entities/Basket.cs
namespace ShopNet.DAL.Entities
{
    //redis entity
    public class Basket
    {
        public string Id { get; set; }
        public List<BasketItem> Items { get; set; } = new List<BasketItem>();

        public Basket()
        {
        }

        public Basket(string id)
        {
            Id = id;
        }
    }

    public class BasketItem
    {
        public int Id { get; set; }
        public string ProductName { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public string PictureUrl { get; set; }
        public ProductBrand Brand { get; set; }
        public ProductType Type { get; set; }
    }
}